docs/evm_compatibility/code_compatibility_checklist.md

```diff
@@ -0,0 +1,82 @@
+
+### Solidity Compiler version
+
+If the contracts are not clean fork like Uniswap V2 or Aave V2 and the solidity compiler is very old i.e, < 0.8.x, it is always recommended to upgrade to latest stable solidity compiler version and re-run the tests.
+:::info
+Solidity version ≥ 0.8.25 is not supported currently because of the need to implement EIP-5656 and EIP-1153. It will be supported on Devnet and Mainnet soon.
+:::
+
+### Usage of third party protocols
+
+There shouldn't be any usage of third party protocols that are not currently supported on Neon EVM.
+
+### Usage of `block.number` and `block.timestamp`
+
+1. `block.timestamp` and `block.number` shouldn't be used as an array or mapping key. More details can be found [here](https://docs.neonevm.org/docs/evm_compatibility/overview#limitation-on-blocktimestamp--blocknumber-usage)
+
+2. `block.timestamp` and `block.number` shouldn't be used as an argument in `create2`(Deterministic deployments) since deterministic addresses are calculated based on the arguments provided.
+
+### Usage of non-reentrancy safe methods `transfer()` and `send()`
+
+`transfer()` and `send()` are not considered as reentrancy safe methods in Neon EVM. This is described [here](https://docs.neonevm.org/docs/evm_compatibility/overview#reentrancy-safe-approaches). It is recommended to use `call()` as an alternative for native token transfers.
+
+```sh
+contract Vulnerable {
+    function withdraw(uint256 amount) external {
+        // This forwards 2300 gas, which may not be enough if the recipient
+        // is a contract and gas costs change.
+        msg.sender.transfer(amount);
+    }
+}
+
+contract Fixed {
+    function withdraw(uint256 amount) external {
+        // This forwards all available gas. Be sure to check the return value!
+        (bool success, ) = msg.sender.call.value(amount)("");
+        require(success, "Transfer failed.");
+    }
+}
+```
+
+### Usage of unsupported OpCodes
+
+There shouldn't be any usage of unsupported opcodes -
+
+1. `gasleft()` or `gas()`
+2. `block.coinbase`
+3. `block.difficulty` / `block.prevrandao`
+4. `block.gaslimit`
+5. `block.basefee`
+6. `selfdestruct` - The opCode behind this method will be deprecated soon. This can be mostly seen with the projects that uses [Seaport marketplace protocol](https://github.com/ProjectOpenSea/seaport/blob/main/contracts/zones/PausableZone.sol#L110). However, if `selfdestruct` is used only for tests/ mocks, this can be skipped.
+
+The details of the unsupported opcodes are described [here](https://docs.neonevm.org/docs/evm_compatibility/opcodes).
+
+### Usage of `multicall` methods
+
+The smart contracts having `multicall` methods that includes more than one address or migration methods to recreate state from another chain results in exceeding the 64 accounts limit. This situation can be avoided by calling these methods in batches.
```

Comment:
@sukanyaparashar please explain here that the 64 accounts limit does not mean 64 addresses on Solidity. We do also have 7 default accounts that used for the transaction execution and this automatically makes a limit of 57 Solidity addresses that can be included in a single transaction. The number can be even smaller if there are more internal calls between contracts.